CW Interface to rigctld

Efpophis opened the pull request:
So I thought, "hey, it's open source, do it yourself!"

So here it is - I have added everything (more or less) needed to key a rig's internal keyer using the existing rigctld connection configured under preferences. I've tested it with my TS-2000 and it works as I would expect - even made a couple QSOs with it.

I noticed that it was impossible to go back and forth from the CW typing window to the newQSO window, so I modified that behavior.

Feel free to suggest any changes, or just grab this and run with it, or whatever. I hope you like it :+1: 

73, 

Bill WK2X

ok2cqr commented:
I'm sorry, I can't use this implementation. uCWKeying unit should be
completely indepentent on cqrlog source code, I use it also in other
projects.

I stated to work on new TCWHamLib class but I don't have any radio
with CW support through hamlib. What are the commands to interrupt CW
sending and tune (CTRL+T in the log), please? I've fount only "b"
command related to morse code but nothing more. Thank you!


2015-04-09 1:35 GMT+02:00 Bill Crossley <notifications@github.com>:
> So I thought, "hey, it's open source, do it yourself!"
>
> So here it is - I have added everything (more or less) needed to key a rig's
> internal keyer using the existing rigctld connection configured under
> preferences. I've tested it with my TS-2000 and it works as I would expect -
> even made a couple QSOs with it.
>
> I noticed that it was impossible to go back and forth from the CW typing
> window to the newQSO window, so I modified that behavior.
>
> Feel free to suggest any changes, or just grab this and run with it, or
> whatever. I hope you like it
>
> 73,
>
> Bill WK2X
>
> ________________________________
>
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/ok2cqr/cqrlog/pull/17
>
> Commit Summary
>
> add hamlib/rigctl CW keying
> Updated changelog
> Made fCWType non-modal so you can type and log at the same time.
>
> File Changes
>
> M src/CHANGELOG (1)
> M src/fCWType.lfm (19)
> M src/fCWType.lrs (58)
> M src/fCWType.pas (6)
> M src/fNewQSO.pas (13)
> M src/fPreferences.lfm (2070)
> M src/fPreferences.lrs (2060)
> M src/fPreferences.pas (5)
> M src/fTRXControl.pas (21)
> M src/uCWKeying.pas (110)
> M src/uRigControl.pas (28)
>
> Patch Links:
>
> https://github.com/ok2cqr/cqrlog/pull/17.patch
> https://github.com/ok2cqr/cqrlog/pull/17.diff
>
> —
> Reply to this email directly or view it on GitHub.



-- 
http://HamQTH.com/ok2cqr
http://ok2cqr.com
http://cqrlog.com
http://cqrtest.com

Efpophis commented:
Hi Petr,

Thanks for looking. I didn't realize the need for uCWKeying to be independent. So that means that instead of using the TRX control form, it will have to have its own TCP connection to rigctld, right? I think I could do that easily enough if you like.

If you have something that needs to be tested, I would be glad to help since my radio does have the support - I've been using this implementation to make several QSO already.

As for the interrupt and tune interface, I don't believe it is supported under Hamlib. However, I did not think it was a big problem since one of the other cw keying classes had this method with only a single comment saying "not supported" as well.

I'll check out your changes for the cwtyping form as soon as I can. Thanks!



-------- Original Message --------
From: Petr Hlozek <notifications@github.com>
Sent: April 12, 2015 6:22:49 AM EDT
To: ok2cqr/cqrlog <cqrlog@noreply.github.com>
Cc: Bill Crossley <efpophis@efpophis.net>
Subject: Re: [cqrlog] CW Interface to rigctld (#17)

I'm sorry, I can't use this implementation. uCWKeying unit should be
completely indepentent on cqrlog source code, I use it also in other
projects.

I stated to work on new TCWHamLib class but I don't have any radio
with CW support through hamlib. What are the commands to interrupt CW
sending and tune (CTRL+T in the log), please? I've fount only "b"
command related to morse code but nothing more. Thank you!


2015-04-09 1:35 GMT+02:00 Bill Crossley <notifications@github.com>:
> So I thought, "hey, it's open source, do it yourself!"
>
> So here it is - I have added everything (more or less) needed to key a rig's
> internal keyer using the existing rigctld connection configured under
> preferences. I've tested it with my TS-2000 and it works as I would expect -
> even made a couple QSOs with it.
>
> I noticed that it was impossible to go back and forth from the CW typing
> window to the newQSO window, so I modified that behavior.
>
> Feel free to suggest any changes, or just grab this and run with it, or
> whatever. I hope you like it
>
> 73,
>
> Bill WK2X
>
> ________________________________
>
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/ok2cqr/cqrlog/pull/17
>
> Commit Summary
>
> add hamlib/rigctl CW keying
> Updated changelog
> Made fCWType non-modal so you can type and log at the same time.
>
> File Changes
>
> M src/CHANGELOG (1)
> M src/fCWType.lfm (19)
> M src/fCWType.lrs (58)
> M src/fCWType.pas (6)
> M src/fNewQSO.pas (13)
> M src/fPreferences.lfm (2070)
> M src/fPreferences.lrs (2060)
> M src/fPreferences.pas (5)
> M src/fTRXControl.pas (21)
> M src/uCWKeying.pas (110)
> M src/uRigControl.pas (28)
>
> Patch Links:
>
> https://github.com/ok2cqr/cqrlog/pull/17.patch
> https://github.com/ok2cqr/cqrlog/pull/17.diff
>
> —
> Reply to this email directly or view it on GitHub.



-- 
http://HamQTH.com/ok2cqr
http://ok2cqr.com
http://cqrlog.com
http://cqrtest.com


---
Reply to this email directly or view it on GitHub:
https://github.com/ok2cqr/cqrlog/pull/17#issuecomment-92030484

Efpophis commented:
I just checked out your implementation, and I like it much better than mine, LOL!

I am going to go ahead and close this PR.  I have one idea for the hamlib CW class that I think might be nice. Perhaps the option to read the initial key speed from hamlib and to poll it periodically instead of strictly setting it from cqrlog?  It would be similar to the "use POT speed" option on the winkeyer interface, I think. I'll play around and see if I can get it to work, and maybe if I can fill in some other unimplemented calls, too.

Thanks for making a GREAT logger!
